### PR TITLE
Extension Management API implementation

### PR DESCRIPTION
**Pull request by GPugnet**
# PR Summary

This is a basic Extension Management API implementation

## PR Checklist

- [x] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [X] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [X] [Update VSTeam.psd1](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#add-a-format-file)
- [ ] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#add-a-format-file)

### CODE REVIEW

**Comment by GPugnet** [comment]
I have a concern regarding the API response. I see on other API already implemented that types are defined, and that the types are applied to the responses. How are defined those types ?

**Comment by DarqueWarrior** [comment]
The types are created by just reviewing the repsonses from the API.  The files are then added and the types applied.  If these types are are going to be used in the provider you can skip the type files and just created classes.  The classes are then used in the provider.

**Comment by DarqueWarrior** [comment]
I am making all the suggested changes.  Let me know if you see anything wrong with my suggestions.

File: src/extensions.psm1 [hunk 1]
```diff
@@ -0,0 +1,161 @@
+Set-StrictMode -Version Latest
+
+# Load common code
+$here = Split-Path -Parent $MyInvocation.MyCommand.Path
+. "$here\common.ps1"
+
+function Add-VSTeamExtension {
```
[the remark this answers is not in the document]
Comment (GPugnet) [reply src/extensions.psm1]: I missed that. Yup, we should add the ShoudlProcess and the Force. I'll dot that soon.

**Pull request merged**